File: python/impl/jazz_plugin.py
from base_plugin import BasePlugin

class JazzPlugin(BasePlugin):
    state: int

    def setup(self):
        print("Relax, setting up...") 
        self.state = 1
    
    def step_1(self):
        print("Fine tunning equalizer")

    def step_2(self):
        print("Playing jazz")

    def teardown(self):
        print("Volume down, stop jazz")

File: typescript/impl/JazzPlugin.ts
import { BasePlugin } from '../BasePlugin';

export class JazzPlugin implements BasePlugin {

    setup(): void{
        console.log("Relax, setting up...");
    }
    step1(): void{
        console.log("Fine tunning equalizer");
    }
    step2(): void{
        console.log("Playing jazz");
    }
    teardown(): void{
        console.log("Volume down, stop jazz");
    }
}